app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -150,25 +217,28 @@ def save_container_routes_inventory(ems, hashes, target = nil)
     store_ids_for_new_records(ems.container_routes, hashes, :ems_ref)
   end
 
-  def save_container_nodes_inventory(ems, hashes, target = nil)
-    return if hashes.nil?
-    target = ems if target.nil?
-
-    ems.container_nodes.reset
-    deletes = if target.kind_of?(ExtManagementSystem)
-                :use_association
-              else
-                []
-              end
-
-    save_inventory_multi(ems.container_nodes, hashes, deletes, [:ems_ref],
-                         [:labels, :tags, :computer_system, :container_conditions,
-                          :additional_attributes], [:namespace])
-    store_ids_for_new_records(ems.container_nodes, hashes, :ems_ref)
+  def graph_container_nodes_inventory(ems, hashes, target = nil)
+    hashes.to_a.each do |h|
+      h = h.except(:labels, :tags, :additional_attributes) # TODO children
+      h = h.except(:namespace)
+      node = @inv_collections[:container_nodes].lazy_find(h[:ems_ref])
```

Comment:
Right, you can use @inv_collections[:container_nodes].lazy_find(h[:ems_ref] with the InventoryObject it points to. The lazy_find will later just become that InventoryObject, with O(1) lookup.

So in AWS I do this to be more readable I am usually using the InventoryObject directly in the case of the child/nested objects. And it's the case that they do arrive together usually, so it should be ok.

If you are not sure about order just use the lazy_find. Or if you plan to build them in a different order later.